File: seminar/src/main/java/com/server/dosopt/seminar/repository/MemberRepository.java
package com.server.dosopt.seminar.repository;

import com.server.dosopt.seminar.domain.Member;
import jakarta.persistence.EntityNotFoundException;
import org.springframework.data.jpa.repository.JpaRepository;

/**
 * 인터페이스에서 구현부를 작성하기 위해 default 키워드를 붙여 메서드 추가
 */
public interface MemberRepository extends JpaRepository<Member, Long> {

    default Member findByIdOrThrow(Long id) {
        return findById(id).orElseThrow(
                () -> new EntityNotFoundException("존재하지 않는 회원입니다.")
        );
    }
}

File: kotlin-seminar/src/main/kotlin/sopt/org/kotlinseminar/repository/MemberRepository.kt
package sopt.org.kotlinseminar.repository

import jakarta.persistence.EntityNotFoundException
import org.springframework.data.jpa.repository.JpaRepository
import sopt.org.kotlinseminar.domain.Member

interface MemberRepository : JpaRepository<Member, Long> {

    fun findByIdOrThrow(id: Long): Member {
        return findById(id).orElseThrow {
            EntityNotFoundException("존재하지 않는 회원입니다.")
        }
    }
}